core/src/com/ericc/the/game/map/LevelFactory.java

```diff
@@ -27,16 +27,22 @@ public static Level generate(int levelNumber) {
             entities.add(new PushableObject(map.getRandomPassableTile(), Media.crate));
         }
 
-        entities.add(new Stairs(map.makeStairs(StaircaseDestination.DESCENDING),
+        Stairs entrance = new Stairs(map.getRandomPassableTileFromRooms(),
+                Media.stairsUp,
+                StaircaseDestination.ASCENDING
+        );
+
+        Stairs exit = new Stairs(map.getRandomPassableTileFromRooms(),
                 Media.stairsDown,
-                StaircaseDestination.DESCENDING)
+                StaircaseDestination.DESCENDING
         );
 
+        entities.add(exit);
+        map.registerStairs(exit);
+
         if (levelNumber > 0) {
-            entities.add(new Stairs(map.makeStairs(StaircaseDestination.ASCENDING),
-                    Media.stairsUp,
-                    StaircaseDestination.ASCENDING)
-            );
+            entities.add(entrance);
```

Comment:
O najs, to jest dużo bardziej eleganckie z tym registerStairs. Ewentualnie samo tworzenie obiektu entrance możesz przesunąć do środka ifa.